win32/tests/job_object_helper_ut/job_object_helper_ut.c

```diff
@@ -0,0 +1,386 @@
+// Copyright(C) Microsoft Corporation.All rights reserved.
+
+
+#include <stdlib.h>
+
+#include "windows.h"
+
+#include "macro_utils/macro_utils.h"
+
+#include "testrunnerswitcher.h"
+#include "umock_c/umock_c.h"
+#include "umock_c/umock_c_negative_tests.h"
+#include "umock_c/umocktypes_windows.h"
+
+#define ENABLE_MOCKS
+    #include "c_pal/interlocked.h"
+    #include "c_pal/interlocked_hl.h"
+    #include "c_pal/gballoc_hl.h"
+    #include "c_pal/gballoc_hl_redirect.h"
+
+    MOCKABLE_FUNCTION(, HANDLE, mocked_CreateJobObject, LPSECURITY_ATTRIBUTES, lpJobAttributes, LPCSTR, lpName);
```

Comment:
>mocked_CreateJobObject [](http://example.com/codeflow?start=32&length=22)

createJobObject and CloseHandle should be paired somehow.

I would personally recomment returning `malloc(1)` from CreateJobObject and then in CloseHandle calling `free` on the argument.

If somehow the calls are not paired, that will result in a vld violation.

Reply:
This is genius.  It revealed a bug if `AssignProcessToJobObject` failed.